src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -103,6 +174,24 @@ public DaprSecretStoreConfigurationProvider(string store, bool normalizeKey, IRe
             this.normalizeKey = normalizeKey;
             this.metadata = metadata;
             this.client = client;
+            this.sidecarWaitDuration = sidecarWaitDuration;
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="normalizeKey">Indicates whether any key delimiters should be replaced with the delimiter ":".</param>
+        /// <param name="keyDelimiters">A collection of delimiters that will be replaced by ':' in the key of every secret.</param>
+        /// <param name="metadata">A collection of metadata key-value pairs that will be provided to the secret store. The valid metadata keys and values are determined by the type of secret store used.</param>
+        /// <param name="client">Dapr client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(
```

Comment:
How many constructors do we absolutely **need** here? Users are going to interact with this type via `AddSecretStore`, really the constructors are just for proper layering. I think it would be fine to just add a single new constructor.

Reply:
Since we're calling it from `DaprSecretStoreConfigurationSource`, which has two distinct paths to creating the Provider, I kept 2 of the constructors. Removed the others which would never be called though.